File: Python/01.ProgrammingBasics/06.Conditional-Statements-Adv-Exercise/09.SkiTrip/_09.SkiTrip.py
days = int(input())
room_type = input()
rating = input()

nights = days - 1
room_price = 0

if room_type == "room for one person":
    room_price = 18.00 * nights
elif room_type == "apartment":
    room_price = 25.00 * nights
    if 10 <= days <= 15:
        room_price *= 0.65
    elif days > 15:
        room_price *= 0.5
elif room_type == "president apartment":
    room_price = 35.00 * nights
    if 10 <= days <= 15:
        room_price *= 0.9
    elif days > 15:
        room_price *= 0.8

if rating == "positive":
    room_price += room_price * 0.25
elif rating == "negative":
    room_price -= room_price * 0.1

print(f"{room_price:.2f}")

File: C++/01-Basics/03.3-ConditionalStatementsAdvЕxercise/09.SkiTrip/09.SkiTrip.cpp
#include <iostream>
#include <iomanip>
#include <string>

using namespace std;

int main() {
    int days;
    string roomType, feedback;
    cin >> days;
    cin.ignore();
    getline(cin, roomType);
    getline(cin, feedback);

    double roomForOnePersonPrice = 18.00;
    double apartmentPrice = 25.00;
    double presidentApartmentPrice = 35.00;

    int nights = days - 1;
    double finalPrice = 0.0;

    if (roomType == "room for one person") {
        finalPrice = nights * roomForOnePersonPrice;
    }
    else if (roomType == "apartment") {
        finalPrice = nights * apartmentPrice;
        if (nights < 10) {
            finalPrice *= 0.70;
        }
        else if (nights <= 15) {
            finalPrice *= 0.65;
        }
        else {
            finalPrice *= 0.50;
        }
    }
    else if (roomType == "president apartment") {
        finalPrice = nights * presidentApartmentPrice;
        if (nights < 10) {
            finalPrice *= 0.90;
        }
        else if (nights <= 15) {
            finalPrice *= 0.85;
        }
        else {
            finalPrice *= 0.80;
        }
    }

    if (feedback == "positive") {
        finalPrice *= 1.25;
    }
    else if (feedback == "negative") {
        finalPrice *= 0.90;
    }

    cout << fixed << setprecision(2) << finalPrice << endl;

    return 0;
}
